### PR TITLE
Add types for @sindresorhus/df npm packge

### PR DESCRIPTION
**Pull request by eightypop**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/sindresorhus__df/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Type definitions for sindresorhus__df 2.10
+// Project: https://github.com/sindresorhus/df (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
```
Comment (andy-ms) [inline types/sindresorhus__df/index.d.ts]: Remove the parenthetical

File: types/sindresorhus__df/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+interface Df {
+    (): Promise<SpaceInfo[]>;
+    fs: (filesystem: string) => Promise<SpaceInfo>;
+    file: (file: string) => Promise<SpaceInfo>;
+}
+
+declare const df: Df;
+
+export default df;
```
Comment (andy-ms) [inline types/sindresorhus__df/index.d.ts]: `require("@sindresorhus/df");` is itself a function, not an object with a `default` property. So this should be `export = df;`.

File: types/sindresorhus__df/sindresorhus__df-tests.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import df from '@sindresorhus/df';
```
Comment (andy-ms) [inline types/sindresorhus__df/sindresorhus__df-tests.ts]: Should be `import df = require('@sindresorhus/df')`, or could leave as-is and add `"esModuleInterop": true` to `tsconfig.json` compilerOptions.

File: types/sindresorhus__df/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+export interface SpaceInfo {
+    filesystem: string;
+    size: number;
+    used: number;
+    available: number;
+    capacity: number;
+    mountpoint: string;
+}
+
+interface Df {
```
Comment (andy-ms) [inline types/sindresorhus__df/index.d.ts]: Declaring an interface to represent the module shape is unnecessary, you can just export a function directly:
```ts
declare function df(): Promise<df.SpaceInfo[]>;
declare namespace df {
    function fs(filesystem: string): Promise<SpaceInfo>;
    function file(file: string): Promise<SpaceInfo>;
    interface SpaceInfo { ... }
}
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: types/sindresorhus__df/index.d.ts (hunk 4 above)
Comment (whatknight) [reply types/sindresorhus__df/index.d.ts]: I'm new to TS. Since the function and namespace are named the same, when we export df they overload?

Comment (andy-ms) [reply types/sindresorhus__df/index.d.ts]: They merge to become a single entity. https://www.typescriptlang.org/docs/handbook/declaration-merging.html

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**